src/pages/api/web3mail/send-mail-to-my-contacts.ts

```diff
@@ -12,6 +13,17 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   if (!privateKey) {
     return res.status(500).json(`Private key is not set`);
   }
+
+  // Check whether the address which provided the signature is the owner of the platform
+  const address = await recoverMessageAddress({
+    message: 'Iexec Web3mail',
+    signature,
+  });
+
+  if (address !== process.env.NEXT_PUBLIC_WEB3MAIL_PLATFORM_ADDRESS) {
```

Comment:
address must be the one owner of the current platform NEXT_PUBLIC_PLATFORM_ID. You can get it from graph